packages/components/autocomplete/Autocomplete.mdx

```diff
@@ -0,0 +1,239 @@
+        // This prop is the function that will tell the component how to render each item in the options list
+        renderItem={(item) => <>{item.name}</>}
+      />
+
+      <Paragraph>Selected fruit: {selectedFruit}</Paragraph>
+    </Stack>
+  );
+}
+```
+
+Both `itemToString` and `renderItem` are necessary when passing objects as items and they both will receive an "item" as an argument.
+
+If you are using Typescript, you can tell the Autocomplete what is the type of your items to make these functions strongly typed.
+You can do that by writing the component like this `<Autocomplete<ItemType> {...props}/>`
+
+### Filling in a list of items.
+
+```jsx
+function AutocompleteExample() {
+  const fruits = [
+    { id: 1, name: 'Apple 🍎' },
+    { id: 2, name: 'Ananas 🍍' },
+    { id: 3, name: 'Avocado 🥑' },
+    { id: 4, name: 'Banana 🍌' },
+    { id: 5, name: 'Coconut 🥥' },
+  ];
+
+  // The state now stores an array of selected fruits, not just a string
+  const [selectedFruits, setSelectedFruits] = React.useState([]);
+
+  const handleFilter = (item, inputValue) => {
+    item.name.toLowerCase().includes(inputValue.toLowerCase());
+  };
+
+  const handleSelectItem = (item) => {
+    // Every time an item is selected, the component will add the name of the fruit to the selectedFruits array
+    setSelectedFruits((prevState) => [...prevState, item.name]);
+  };
+
+  return (
+    <Stack flexDirection="column" alignItems="start">
+      <Autocomplete
+        items={fruits}
+        onFilter={handleFilter}
+        onSelectItem={handleSelectItem}
+        itemToString={(item) => item.name}
+        renderItem={(item) => <>{item.name}</>}
+        // When this prop is `true`, it will clean the TextInput after an option is selected
+        clearAfterSelect
+      />
+
+      <span>
+        <Paragraph>Selected fruits:</Paragraph>
+        <ul>
+          {selectedFruits.map((fruit, index) => (
+            <li key={index}>{fruit}</li>
+          ))}
+        </ul>
+      </span>
+    </Stack>
+  );
+}
+```
+
+### Error validation with FormControl
+
+```jsx
+function AutocompleteExample() {
+  const fruits = [
+    'Apple 🍎',
+    'Ananas 🍍',
+    'Avocado 🥑',
+    'Banana 🍌',
+    'Coconut 🥥',
+  ];
+
+  const [selectedFruit, setSelectedFruit] = React.useState();
+  const [isInvalid, setIsInvalid] = React.useState(false);
+
+  const handleFilter = (item, inputValue) => {
+    item.toLowerCase().includes(inputValue.toLowerCase());
+  };
+
+  const handleSelectItem = (item) => {
+    setSelectedFruit(item);
+  };
+
+  return (
+    <Stack flexDirection="column" alignItems="start">
+      <FormControl isInvalid={isInvalid}>
+        <FormControl.Label>Favorite fruit:</FormControl.Label>
+
+        <Autocomplete
+          items={fruits}
+          onFilter={handleFilter}
+          onSelectItem={handleSelectItem}
+        />
+
+        {isInvalid && (
+          <FormControl.ValidationMessage>Error</FormControl.ValidationMessage>
```

Comment:
Unrelated to this PR, but this component renders paragraph with `aria-live="polite"` as it requires the user immediate attention, i think it should be `aria-live="assertive"`

Reply:
@mshaaban0 do you mind opening a PR for validation message component?

Reply:
created quick fix https://github.com/contentful/forma-36/pull/1350

Reply:
Ohhhh good catch! I will do it in another PR

Reply:
I guess I replied to moe before refreshing the page hihihi